content/qemu/index.md

```diff
@@ -3,40 +3,97 @@ title = 'QEMU'
 icon = 'clanbomber'
 +++
 
-<a href="https://www.qemu.org/">QEMU</a> is an emulator for
-various CPUs. It works on Linux, Windows, FreeBSD and Mac OS X. It's possible
-to run grml with QEMU. Notice that it's much slower than running it native
-due to the emulation mode.
+[QEMU](https://www.qemu.org/) is an emulator for various CPUs.
+It works on Linux, Windows, FreeBSD and Mac OS X.
+It's possible to run grml with QEMU.
+Note that it is slower than running it native due to the emulation mode.
 
 ## Using QEMU with acceleration
 
-grml >=0.5 provides the <a href="https://www.qemu.org/">accelerator module</a> for QEMU namend 'kqemu'. Just run the following commands to use it:
+On modern Linux distributions, including Grml, the kernel comes with an acceleration module, called `KVM`.
+To use this, load the `kvm.ko` kernel module, and give yourself rights:
 
 <pre class="rahmen">
-apt-get update ; apt-get install qemu # install qemu
-modprobe kqemu                        # load the kernel module
-mknod /dev/kqemu c 250 0              # create the device
-chmod 666 /dev/kqemu /dev/net/tun     # adjust permissions
+apt update && apt install qemu-system        # install qemu
+modprobe kvm                                 # load the kernel module
+adduser $USER kvm                            # add your user to kvm group
+# now login again or try: newgrp kvm
 </pre>
 
-To see if kqemu is enabled and working correctly, use the QEMU monitor command (press ctrl-alt-2): 'info kqemu'
+To see if kvm is enabled and working correctly, use the QEMU monitor command (press ctrl-alt-2): 'info kvm'
 
-## Problems with Qemu
+## Usage examples
 
-### Black / blank screen
+In all examples below you can add `-nographic` to get the first serial console on your terminal.
+This might work better on arm64, where most software defaults to supporting serial console, and less good on amd64.
```

Comment:
Hmm, what do you mean with "most software defaults to supporting serial console" on `arm64`? Why is that so?

Reply:
EDK2, grub, linux kernel appear to default-enable serial console support. This is probably because there is no native "VGA" like on x86.

ISTR on x86 most of these things don't bother enabling serial.

Reply:
Ah, ok...
I haven't tested Grml/arm64 with -nographic on my system, but Grml/amd64 does not give me the serial console. It would be nice to let Grub detect that somehow... is that even possible?

Reply:
grub: good question! it does somehow work on arm64, so i guess it should be possible on x86?

Reply:
I've done my tests on arm64 with the arm64 ISO, and `-nographic` just works ootb.

Reply:
Ah... with `qemu-system-x86_64 -M q35 -m 2048 -cpu host -enable-kvm -nographic -cdrom /srv/isos/grml-full-2024.12-amd64.iso` I get:
```
SeaBIOS (version 1.16.2-debian-1.16.2-1)


iPXE (http://ipxe.org) 00:02.0 CA00 PCI2.10 PnP PMM+7EFCEDF0+7EF0EDF0 CA00
                                                     


Booting from Hard Disk...
Boot failed: could not read the boot disk

Booting from DVD/CD...

ISOLINUX 6.04 20200816 ETCD Copyright (C) 1994-2015 H. Peter Anvin et al
```

If I hit ESC within the timeout of 25s I get the boot prompt where I can enter `serial`. If not I am stuck...

Can you confirm that behavior? (Maybe worth to be documented somehow?)

Reply:
> ISTR on x86 most of these things don't bother enabling serial.

Looks like at least grub enables serial, if I remove VGA. Kernel doesn't automatically enable serial though. Might be worth investigating if that can be `if`-ed in grub.

Reply:
> Can you confirm that behavior? (Maybe worth to be documented somehow?)

Yes, I can. The trick appears to add this to the qemu command line: `-vga none`

Then ISOLINUX presents this menu:

<img width="543" alt="Screenshot 2024-12-24 at 00 00 10" src="https://github.com/user-attachments/assets/dea4636e-26dc-4b07-996f-b26f2eb5e506" />